lms/static/coffee/src/instructor_dashboard/membership.coffee

```diff
@@ -670,6 +670,564 @@ class AuthList
+      problems = problems.slice(0,-1)
+    send_data =
+      filter: @filtering
+      entityName: @entityName
+    $.ajax
+      dataType: 'json'
+      url: @use_query_endpoint
+      data: send_data
+      success: (data) -> cb? null, data
+      error: std_ajax_err ->
+        cb? gettext("Error getting students")
+
+  #make a single query to the backend. doesn't wait for
+  #query completion as that can take awhile
+  reload_students: (tr) ->
+    @$save_query_btn.addClass("disabled")
+    @$email_csv_btn.addClass("disabled")
+    @$email_csv_btn[0].value = "Aggregating Queries"
+    tr.addClass("working")
+    @get_students (error, students) =>
+      if error
+        $broken_icon = $ _.template('<div class="done">
+          <i class="icon-warning-sign"></i> <%= label %></div>',
+          {label: "Sorry, we're having a problem with this query.
+            Please delete this row and try again."})
+        tr.children()[4].innerHTML = $broken_icon[0].outerHTML
+      return @show_errors error unless error is null
+
+  #we don't care if these calls succeed or not so no wrapped callback
+  delete_temp_query: (queryId)->
+    send_url = @$delete_temp_endpoint + "/" + queryId
+    $.ajax
+      dataType: 'json'
+      url: send_url
+
+  delete_bulk_temp_query: (queryIds)->
+    send_url = @$delete_bulk_temp_endpoint
+    send_data =
+      existing: queryIds.join(',')
+    $.ajax
+      dataType: 'json'
+      url: send_url
+      data: send_data
+
+  delete_saved_query: (queryId)->
+    send_url = @$delete_saved_endpoint + "/" + queryId
+    $.ajax
+      dataType: 'json'
+      url: send_url
+
+  #adds a row to active queries
+  start_row:(color, arr, rowIdClass, table) ->
+    #find which row to insert in
+    idx =0
+    orIdx = 0
+    andIdx = 0
+    notIdx = 0
+    useIdx = 0
+    rows = table[0].children
+    #figuring out where to place the new row
+    #we want the group order to be and, or, not
+    for curRow in rows
+      idx += 1
+      if curRow.classList.contains("or")
+        orIdx = idx
+      if curRow.classList.contains("and")
+        andIdx = idx
+      if curRow.classList.contains("not")
+        notIdx = idx
+      if curRow.classList.contains(color)
+        useIdx = idx
+    if color == "or" and useIdx == 0
+      useIdx = Math.max(notIdx, andIdx)
+    if color == "not" and useIdx == 0
+      useIdx =andIdx
+    row = table[0].insertRow(useIdx)
+    if rowIdClass.hasOwnProperty('id')
+      row.id = rowIdClass['id']
+    if rowIdClass.hasOwnProperty('query')
+      row.setAttribute('query',rowIdClass['query'])
+
+    row.classList.add(color.toLowerCase())
+    if rowIdClass.hasOwnProperty('class')
+      _.each rowIdClass['class'], (addingClass) ->
+        row.classList.add(addingClass.toLowerCase())
+    for num in [0..3]
+      cell = row.insertCell(num)
+      item = arr[num]
+      cell.innerHTML = item['text']
+      if item.hasOwnProperty('id') and item['id'] !=""
```

Comment:
`hasOwnProperty` is import to use when iterating over an object with the form `for (var property in object) { ... }`, but I believe it's superfluous here. In fact, I think we can shorten this whole thing to:
```javascript
cell.id = item['id'] || cell.id
```